File: code_combat/7_Sarven_Desert/304-Canyon_of_Storms/canyonOfStorms.js
// A desert storm is coming!
// Yaks can detect when a storm is coming.

// This variable will be used as a condition.
var yak = hero.findNearestEnemy();

// While there is at least one sand yak:
while (yak) {
    var item = hero.findNearestItem();
    if (item) {
        hero.moveXY(item.pos.x, item.pos.y);
    }
    // Update the value of the variable `yak`
    // with findNearestEnemy()
    yak = hero.findNearestEnemy();
}
// The yaks have hidden.
// Move to the red X at the hideout.
hero.moveXY(38, 58);

File: projects/code_combat/7_Sarven_Desert/304-Canyon_of_Storms/canyon_of_storms.py
yak = hero.findNearestEnemy()

while yak:
    item = hero.findNearestItem()
    if item:
        hero.moveXY(item.pos.x, item.pos.y)
    yak = hero.findNearestEnemy()

hero.moveXY(38, 58)
